viewing_party/party.py

```diff
@@ -1,23 +1,224 @@
+        if movies["title"] == title: 
+            user_data["watchlist"].remove(movies)
+            user_data["watched"].append(movies)
+    return user_data
+
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+## Create a function that will check the average rating of movies watched
+## if average is Zero, return error message
+
+'''
+input: a dictionary called 'user_data' with a 'watched' list of
+movie dictionaries (movies that user has watched)
+-- Calculate average rating of all movies user has watched
+-- An empty watch list has an average rating of 0.0
+output: the average rating
+# {   'watched': [   {   'genre': 'Fantasy',
+#                        'rating': 4.8,
+#                        'title': 'The Lord of the Functions: The Fellowship of '
+#                                 'the Function'},
+#                    {   'genre': 'Fantasy',
+#                        'rating': 4.0,
+#                        'title': 'The Lord of the Functions: The Two '
+#                                 'Parameters'},
+#                    {   'genre': 'Fantasy',
+#                        'rating': 4.0,
+#                        'title': 'The Lord of the Functions: The Return of the '
+#                                 'Value'},
+#                    {   'genre': 'Action',
+#                        'rating': 2.2,
+#                        'title': 'The JavaScript and the React'},
+#                    {'genre': 'Intrigue', 'rating': 2.0, 'title': 'Recursion'},
+#                    {   'genre': 'Intrigue',
+#                        'rating': 4.5,
+#                        'title': 'Instructor Student TA Manager'}]}
+'''
+
+def get_watched_avg_rating(user_data):
+    watched_movies = user_data["watched"] ## array of dictionaries
+    total_rating = []
+    counter = 0 
+## for a dictionary in the list of movies
+    for movie_data in watched_movies:
+        v = movie_data["rating"]
+        total_rating.append(v)
+    if len(total_rating) == 0:
+        return 0.0
+    average = sum(total_rating) / len(total_rating)
+    return average
+
+
+
+## def function to track most watched genre
+## COUNT of times genre appears
+## If there is no amount, return error message
+# '''
+# input: take in the parameter 'user_data' : a nested dictionary 
+# with a VALUE 'watched' list of movie dictionaries
+# each movie has the KEY "genre"
+# >> this VALUE represents user's list of watched movies
+# >> each watched movie contains a genre
+# >>>> the VALUES of  KEY "genre" is a string
+# create a COUNTER to count which "genre" occurs most often
+# If the value of "watched" is empty, return None
 
+# output: The string of the genre that is most frequently watched,
+# if value of watched is an empty list could try ... else none
 
+# '''
+
+def get_most_watched_genre(user_data):
+    watched_movies = user_data["watched"]
+    popular_genre = []

```

Comment:
I think it would be a good exercise to look at how this could be implemented differently with a dictionary instead of a list. What if we built up a mapping using genres for keys with their counts for values?